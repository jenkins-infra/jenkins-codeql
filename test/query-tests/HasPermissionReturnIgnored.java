import hudson.security.AccessControlled;
import hudson.security.Permission;

public class HasPermissionReturnIgnored {
    public void thisIsWrong(AccessControlled ac) throws Exception {
        ac.hasPermission(Permission.READ);
    }

    public void thisIsOK(AccessControlled ac) throws Exception {
        if (!ac.hasPermission(Permission.READ)) {
            throw new Exception("prohibited!");
        }
    }

    public void thisIsAlsoOK(AccessControlled ac) throws Exception {
        ac.checkPermission(Permission.READ);
    }
}