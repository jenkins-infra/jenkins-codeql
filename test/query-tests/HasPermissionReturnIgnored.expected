| HasPermissionReturnIgnored.java:6:9:6:41 | hasPermission(...) | The result of the call is ignored |
